mobile/lib/pages/editing/edit.page.dart

```diff
@@ -0,0 +1,107 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:fluttertoast/fluttertoast.dart';
+import 'package:immich_mobile/pages/editing/crop.page.dart';
+import 'package:immich_mobile/entities/asset.entity.dart';
+import 'package:immich_mobile/widgets/common/immich_image.dart';
+import 'package:immich_mobile/widgets/common/immich_toast.dart';
+@immutable
+class EditImagePage extends StatelessWidget {
+  final Asset? asset;
+  final Image? image;
+  const EditImagePage({
+    super.key,
+    this.image,
+    this.asset,
+  }) : assert((image != null && asset == null) || (image == null && asset != null), 'Must supply one of asset or image');
+  ImageProvider _getImageProvider() {
+    if (asset is Asset) {
+      return ImmichImage.imageProvider(asset: asset);
+    } else if (image is Image) {
+      return (image as Image).image;
+    } else {
+      throw Exception('Invalid image source type');
+    }
+  }
+  void _navigateToCropImagePage(BuildContext context) {
+    Navigator.push(
+      context,
+      MaterialPageRoute(
+        builder: (context) => CropImagePage(image: _getImageWidget()),
+      ),
+    );
+  }
+  Image _getImageWidget() {
+    if (asset != null) {
+      return Image(image: ImmichImage.imageProvider(asset: asset));
+    } else if (image != null) {
+      return image!;
+    } else {
+      throw Exception('Invalid image source type');
+    }
+  }
+  @override
+  Widget build(BuildContext context) {
+    final ImageProvider provider = _getImageProvider();
+
```

Comment:
You can also add a test condition here:

```dart

if (asset == null && image == null) {
  return SomeKindOfErrorWidget();
}
```

Just in case the `assert` in the constructor wasn't enough. It may also inform the compiler that at least one of them is not null, and then we don't need to do as many `!` statements.

Reply:
I integrated that into the provider variable